File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    ans = 1
    for i in range(1, (yellow // 2) + 1):
        k = yellow / i
        if brown - (2 * (i+2)) == k * 2:
            ans = i
            break
    # print(ans)
    return [(yellow / ans) + 2, ans + 2]

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    if ((yellow * 2) + 6 === brown){
        return [yellow + 2, 3];  
    }
    let h = 2, w = 0;
    
    while(h<1500){
        if (yellow % h === 0){
            w = yellow / h;
            if (((w*2) + (h+2)*2) === brown){
                return [w+2, h+2]
            }
        } 
        h++;
    }
}